tests/sentry/auth/test_staff.py

```diff
@@ -197,6 +206,58 @@ def test_logout_clears_session(self):
         assert not staff.is_active
         assert not request.session.get(SESSION_KEY)
 
+    def test_middleware_as_staff(self):
```

Comment:
It'll be good to have some end to end tests. Something like:
https://github.com/getsentry/sentry/blob/31ad746d283dc367a3191c50fe3c19acccdea7e7/tests/sentry/middleware/test_subdomain.py#L94

and you can inject the new middleware into the middleware stack like this: https://github.com/getsentry/sentry/blob/31ad746d283dc367a3191c50fe3c19acccdea7e7/tests/sentry/middleware/test_customer_domain.py#L205-L210